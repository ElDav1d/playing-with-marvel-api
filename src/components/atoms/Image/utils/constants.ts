import { Thumbnail } from '@/components/pages/Characters/interfaces/characters';
import { PicVariantWidths, PicVariantHeights } from '../interfaces/image';

/* eslint-disable camelcase */
const PIC_VARIANT_WIDTHS: PicVariantWidths = {
  standard_small: 65,
  standard_xlarge: 200,
  standard_fantastic: 250,
  landscape_amazing: 250,
  landscape_incredible: 464,
};
/* eslint-enable camelcase */

/* eslint-disable camelcase */
const PIC_VARIANT_HEIGHTS: PicVariantHeights = {
  standard_small: 45,
  standard_xlarge: 200,
  standard_fantastic: 250,
  landscape_amazing: 156,
  landscape_incredible: 261,
};
/* eslint-enable camelcase */

const PIC_FALLBACK: Thumbnail = {
  path: 'http://i.annihil.us/u/prod/marvel/i/mg/b/40/image_not_available',
  extension: 'jpg',
};

const LAZYLOAD_SIZING_DEFAULT = 'standard_small';

export { PIC_VARIANT_WIDTHS, PIC_VARIANT_HEIGHTS, PIC_FALLBACK, LAZYLOAD_SIZING_DEFAULT };
