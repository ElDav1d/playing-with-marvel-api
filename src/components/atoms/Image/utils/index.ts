export { default as orderSizing } from './orderSizing';
export { default as getSrc } from './getSrc';
export { default as getSrcSet } from './getSrcSet';
export { default as getSizes } from './getSizes';
