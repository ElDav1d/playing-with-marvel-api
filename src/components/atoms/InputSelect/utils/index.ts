export { default as overrideClasses } from './overrideClasses';
