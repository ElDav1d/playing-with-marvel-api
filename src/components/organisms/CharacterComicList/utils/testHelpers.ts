import mockCharacterComics from '../mocks/mockCharacterComics.json';
import userEvent from '@testing-library/user-event';
import { useCharacterComics } from '../hooks';

const mockUseCharacterComics = useCharacterComics as jest.Mock;

export const setUpCharacterComics = () => {
  mockUseCharacterComics.mockReturnValue({
    comics: mockCharacterComics,
    totalComics: 0,
    rangeInit: 0,
    rangeEnd: 0,
    isErrorOnComics: false,
    isFetchingComics: false,
    isFirstPage: true,
    isLastPage: true,
    refetch: jest.fn(),
  });
};

export const setUpHappyPathWithUser = () => {
  setUpCharacterComics();
  return { user: userEvent.setup() };
};
