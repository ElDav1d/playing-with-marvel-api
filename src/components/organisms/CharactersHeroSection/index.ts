export { default as CharactersHeroSection } from './CharactersHeroSection';
