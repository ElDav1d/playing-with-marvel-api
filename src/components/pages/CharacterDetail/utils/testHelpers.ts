import mockCharacterDetail from '../mocks/mockCharacterDetail.json';
import mockCharacterComics from '../mocks/mockCharacterComics.json';
import { useCharacterDetails } from '../hooks';
import userEvent from '@testing-library/user-event';
import { useCharacterComics } from '@/components/organisms/CharacterComicList/hooks';

const mockUseCharacterDetails = useCharacterDetails as jest.Mock;

const mockUseCharacterComics = useCharacterComics as jest.Mock;

export const setUpCharacterDetails = () => {
  mockUseCharacterDetails.mockReturnValue({
    isLoadingCharacter: false,
    isErrorOnCharacter: false,
    character: mockCharacterDetail,
  });
};

export const setUpCharacterComics = () => {
  mockUseCharacterComics.mockReturnValue({
    comics: mockCharacterComics,
    totalComics: 0,
    rangeInit: 0,
    rangeEnd: 0,
    isErrorOnComics: false,
    isFetchingComics: false,
    isFirstPage: true,
    isLastPage: true,
    refetch: jest.fn(),
  });
};

export const setUpHappyPath = () => {
  setUpCharacterDetails();
  setUpCharacterComics();
};

export const setUpHappyPathWithUser = () => {
  setUpHappyPath();
  return { user: userEvent.setup() };
};
