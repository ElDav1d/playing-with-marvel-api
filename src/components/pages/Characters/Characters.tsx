import { ChangeEvent, useEffect, useRef, useState, useMemo } from 'react';
import { useInView } from 'react-intersection-observer';
import { useCharacters, useDebounce } from './hooks';
import { CharacterItem, FetchingOrder, FilterCriteria } from './interfaces/characters';
import CharactersList from '@/components/organisms/CharactersList/CharactersList';
import {
  SELECT_ORDER_LITERALS,
  EMPTY_DATA_LITERAL_LIST,
  LOADER_SIZE,
  MARVEL_RED,
  MAX_FETCH_CHARACTERS_DEFAULT,
  MAX_FETCH_CHARACTERS_OPTIM,
  MAX_FETCH_CHARACTERS_TOP,
  REGEX_IMAGE_PATH,
} from '@/utils/constants';
import Header from '@/components/organisms/Header';
import Footer from '@/components/organisms/Footer';
import Container from '@/components/organisms/Container';
import { RingLoader } from 'react-spinners';
import SideDrawer from '@/components/organisms/SideDrawer';
import CharactersControlPanel from '@/components/organisms/CharactersControlPanel';

const Characters = () => {
  const [searchInput, setSearchInput] = useState<string>('');
  const [searchString, setSearchString] = useState<string>('');
  const [order, setOrder] = useState<FetchingOrder>(FetchingOrder.NAME_AZ);
  const [onClearData, setOnClearData] = useState(false);
  const [filters, setFilters] = useState<FilterCriteria[]>([]);
  const [onClearFilters, setOnClearFilters] = useState(false);
  const maxCharactersRef = useRef(MAX_FETCH_CHARACTERS_DEFAULT);

  const { isError, characters, fetchNextPage, hasNextPage, isFetching, isFetchingNextPage } =
    useCharacters({ maxCharacters: maxCharactersRef.current, searchString, order, onClearData });

  const { ref, inView } = useInView({
    threshold: 0.1,
  });

  useEffect(() => {
    if (inView) {
      fetchNextPage();
    }
  }, [inView]);

  useEffect(() => {
    if (onClearData) {
      setOnClearData(false);
    }
  }, [order, searchString]);

  useDebounce(searchInput, 500, () => setSearchString(searchInput));

  const filterLiterals = ['With Image', 'With Description'];

  const filteredCharacters = useMemo(() => {
    const hasImage = (path: string) => !REGEX_IMAGE_PATH.test(path);
    const hasDescription = (description: string) => description && description !== ' ';

    let maxCharacters;
    let filterCallback;

    switch (true) {
      case filters.includes(FilterCriteria.IMAGE) && filters.includes(FilterCriteria.DESCRIPTION):
        maxCharacters = MAX_FETCH_CHARACTERS_TOP;
        filterCallback = (character: CharacterItem) =>
          hasImage(character.thumbnail.path) && hasDescription(character.description);
        break;

      case filters.includes(FilterCriteria.IMAGE):
        maxCharacters = MAX_FETCH_CHARACTERS_OPTIM;
        filterCallback = (character: CharacterItem) => hasImage(character.thumbnail.path);
        break;

      case filters.includes(FilterCriteria.DESCRIPTION):
        maxCharacters = MAX_FETCH_CHARACTERS_OPTIM;
        filterCallback = (character: CharacterItem) => hasDescription(character.description);
        break;

      default:
        maxCharactersRef.current = MAX_FETCH_CHARACTERS_DEFAULT;
        return characters;
    }

    maxCharactersRef.current = maxCharacters;
    return characters.filter(filterCallback);
  }, [characters, filters]);

  const orderHandler = (event: ChangeEvent<HTMLSelectElement>): void => {
    const value = event.target.value as FetchingOrder;

    setOnClearData(true);
    setOrder(value);
  };

  const hasListDefaults =
    searchInput === '' && order === FetchingOrder.NAME_AZ && filters.length === 0;

  const handleClear = () => {
    setSearchInput('');
    setSearchString('');
    setOrder(FetchingOrder.NAME_AZ);
    setFilters([]);
    setOnClearFilters(true);
    setOnClearData(true);
  };

  const handleClearChecks = () => {
    setOnClearFilters(false);
  };

  return (
    <>
      <Header classNameHeader='flex justify-center'>
        <SideDrawer classNameContainer='bg-black'>
          <CharactersControlPanel
            searchInput={searchInput}
            setSearchInput={setSearchInput}
            searchTitle={'Search by name'}
            searchPlaceholder={'Type a character name'}
            setOnClearData={setOnClearData}
            isEmptyData={!isFetching && filteredCharacters?.length === 0}
            emptyDataLiteral={EMPTY_DATA_LITERAL_LIST}
            orderTitle='Order results'
            onOrderChange={(event) => orderHandler(event)}
            orderOptions={Object.values(FetchingOrder)}
            orderLiterals={SELECT_ORDER_LITERALS}
            filtersTitle='Filter results:'
            filtersOptions={Object.values(FilterCriteria)}
            filtersLiterals={filterLiterals}
            setFilters={setFilters}
            setOnClearChecks={handleClearChecks}
            onClearChecks={onClearFilters}
          />
        </SideDrawer>
      </Header>
      <Container element={'main'}>
        <section
          className='relative z-0 bg-black py-10 mb-4 bg-hero-image bg-center bg-cover
        before:absolute  before:content[""] before:h-full before:w-full  before:bg-gradient-to-b before:from-trans-0.75-black before:from-50% before:left-0 before:top-0 before:z-[-1]'
        >
          <Container element={'div'}>
            <div className='w-full text-white text-center mb-4'>
              <h2 className='text-2xl font-semibold text-white text-center uppercase mb-1'>
                Marvel Characters
              </h2>
              <p className='text-sm text-center'>
                Get hooked on a hearty helping of heroes and villains from the humble House of
                Ideas!
              </p>

              {!hasListDefaults && (
                <>
                  <p className='text-sm mb-2'>
                    Results for
                    {searchInput && <strong> {searchInput}</strong>}
                    {filters && <strong> {filters}</strong>}
                    {order && <strong> {order}</strong>}
                  </p>
                  <button
                    className='text-sm border border-white p-1 active-border focus-visible-border'
                    onClick={handleClear}
                  >
                    CLEAR
                  </button>
                </>
              )}
            </div>
            <CharactersControlPanel
              isDesktop
              searchInput={searchInput}
              setSearchInput={setSearchInput}
              searchTitle={'Search by name'}
              searchPlaceholder={'Type a character name'}
              setOnClearData={setOnClearData}
              isEmptyData={!isFetching && filteredCharacters?.length === 0}
              emptyDataLiteral={EMPTY_DATA_LITERAL_LIST}
              orderTitle='Order results'
              onOrderChange={(event) => orderHandler(event)}
              orderOptions={Object.values(FetchingOrder)}
              orderLiterals={SELECT_ORDER_LITERALS}
              filtersTitle='Filter results:'
              filtersOptions={Object.values(FilterCriteria)}
              filtersLiterals={filterLiterals}
              setFilters={setFilters}
              setOnClearChecks={handleClearChecks}
              onClearChecks={onClearFilters}
            />
          </Container>
        </section>
        {isError && <h2>Oooops, there&apos;s an unexpected error...try reloading again!</h2>}

        {isFetching && !isFetchingNextPage && (
          <RingLoader
            color={MARVEL_RED}
            size={LOADER_SIZE}
            className='mx-auto my-6'
            role='alert'
            aria-label='Characters List is loading'
            aria-busy='true'
            aria-live='polite'
          />
        )}

        {!isFetching && (
          <Container>
            {filteredCharacters?.length === 0 ? (
              <h3 className='text-center'>{EMPTY_DATA_LITERAL_LIST}</h3>
            ) : (
              <CharactersList characters={filteredCharacters} />
            )}
          </Container>
        )}

        {hasNextPage && (
          <div ref={ref}>
            <RingLoader
              color={MARVEL_RED}
              size={LOADER_SIZE}
              className='mx-auto my-6'
              role='alert'
              aria-label='Characters List is loading'
              aria-busy='true'
              aria-live='polite'
            />
          </div>
        )}
      </Container>
      <Footer />
    </>
  );
};

export default Characters;
