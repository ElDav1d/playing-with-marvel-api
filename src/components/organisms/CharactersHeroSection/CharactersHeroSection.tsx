import { CharactersControlPanelInfo } from '@/components/molecules/CharactersControlPanelInfo';
import Container from '@/components/organisms/Container';
import { useMediaQuery } from '@/hooks';
import { HERO_PARAGRAPH_LITERAL, HERO_TITLE_LITERAL, MEDIA_BREAKPOINTS } from '@/utils/constants';
import CharactersControlPanel from '../CharactersControlPanel';

const CharactersHeroSection = () => {
  const isDesktop = useMediaQuery(`(min-width: ${MEDIA_BREAKPOINTS.MD}px)`);

  return (
    <section
      className='relative flex flex-col items-center justify-around pt-logoDefaultHeight gap-2 min-h-64 md:min-h-96 lg:min-h-72 mb-4 bg-black bg-hero-image bg-center bg-cover
    before:absolute before:content[""] before:bg-gradient-to-b before:from-trans-0.5-black before:from-50% before:left-0 before:top-0 before:z-[-1]'
    >
      <Container element='div' className='my-4 lg:w-2/3 flex flex-col gap-4'>
        <div className='text-white text-center'>
          <h2 className='text-2xl md:text-3xl font-semibold text-white text-center uppercase mb-1'>
            {HERO_TITLE_LITERAL}
          </h2>
          <p className='text-sm md:text-base text-center'>{HERO_PARAGRAPH_LITERAL}</p>
        </div>
        {isDesktop && <CharactersControlPanel isDesktop />}
        <CharactersControlPanelInfo />
      </Container>
    </section>
  );
};

export default CharactersHeroSection;
