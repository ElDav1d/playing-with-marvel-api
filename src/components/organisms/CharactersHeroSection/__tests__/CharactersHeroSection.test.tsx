import userEvent from '@testing-library/user-event';
import { render, screen, waitFor } from '@testing-library/react';
import CharactersHeroSection from '../CharactersHeroSection';
import { CharactersProvider } from '@/components/pages/Characters/context';
import { QueryClient, QueryClientProvider } from '@tanstack/react-query';
import { setUpMatchMedia } from '@/utils/testHelpers';
import { act } from 'react-dom/test-utils';
import { DEBOUNCE_DELAY } from '@/utils/constants';

const queryClient = new QueryClient();

it('renders a title', () => {
  // ARRANGE
  setUpMatchMedia(true);

  // ACT
  render(
    <QueryClientProvider client={queryClient}>
      <CharactersProvider>
        <CharactersHeroSection />
      </CharactersProvider>
    </QueryClientProvider>,
  );

  const title = screen.getByRole('heading', { level: 2, name: /characters/i });

  // ASSERT
  expect(title).toBeInTheDocument();
});

it('renders a control panel form when desktop', () => {
  // ARRANGE
  setUpMatchMedia(true);

  // ACT
  render(
    <QueryClientProvider client={queryClient}>
      <CharactersProvider>
        <CharactersHeroSection />
      </CharactersProvider>
    </QueryClientProvider>,
  );

  const controlPanel = screen.getByRole('form', { name: /control panel/i });

  // ASSERT
  expect(controlPanel).toBeInTheDocument();
});

it('matches desktop snapshot', () => {
  // ARRANGE
  setUpMatchMedia(true);

  // ACT
  const { asFragment } = render(
    <QueryClientProvider client={queryClient}>
      <CharactersProvider>
        <CharactersHeroSection />
      </CharactersProvider>
    </QueryClientProvider>,
  );

  // ASSERT
  expect(asFragment()).toMatchSnapshot();
});

it('omits the control panel form when mobile', () => {
  // ARRANGE
  setUpMatchMedia(false);

  // ACT
  render(
    <QueryClientProvider client={queryClient}>
      <CharactersProvider>
        <CharactersHeroSection />
      </CharactersProvider>
    </QueryClientProvider>,
  );

  const controlPanel = screen.queryByRole('form', { name: /control panel/i });

  // ASSERT
  expect(controlPanel).not.toBeInTheDocument();
});

it('matches mobile snapshot', () => {
  // ARRANGE
  setUpMatchMedia(false);

  // ACT
  const { asFragment } = render(
    <QueryClientProvider client={queryClient}>
      <CharactersProvider>
        <CharactersHeroSection />
      </CharactersProvider>
    </QueryClientProvider>,
  );

  // ASSERT
  expect(asFragment()).toMatchSnapshot();
});

it('renders a search input', () => {
  // ARRANGE
  setUpMatchMedia(true);

  // ACT
  render(
    <QueryClientProvider client={queryClient}>
      <CharactersProvider>
        <CharactersHeroSection />
      </CharactersProvider>
    </QueryClientProvider>,
  );

  const searchInput = screen.getByRole('textbox', { name: /search/i });

  // ASSERT
  expect(searchInput).toBeInTheDocument();
});

it('renders a fully functional clear button when search has a value', async () => {
  // ARRANGE
  const user = userEvent.setup();

  setUpMatchMedia(true);

  jest.useFakeTimers();

  // ACT
  render(
    <QueryClientProvider client={queryClient}>
      <CharactersProvider>
        <CharactersHeroSection />
      </CharactersProvider>
    </QueryClientProvider>,
  );

  const searchInput = screen.getByRole('textbox', { name: /search/i });

  user.type(searchInput, 'test');

  act(() => {
    jest.advanceTimersByTime(DEBOUNCE_DELAY);
  });

  await waitFor(() => {
    const clearButton = screen.getByRole('button', { name: /clear/i });

    user.click(clearButton);
  });

  // ASSERT
  await waitFor(() => {
    expect(searchInput).toHaveValue('');
  });
});
