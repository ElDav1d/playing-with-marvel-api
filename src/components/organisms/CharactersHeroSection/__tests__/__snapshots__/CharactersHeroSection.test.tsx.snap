// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches desktop snapshot 1`] = `
<DocumentFragment>
  <section
    class="relative flex flex-col items-center justify-around pt-logoDefaultHeight gap-2 min-h-64 md:min-h-96 lg:min-h-72 mb-4 bg-black bg-hero-image bg-center bg-cover before:absolute before:content[\\"\\"] before:bg-gradient-to-b before:from-trans-0.5-black before:from-50% before:left-0 before:top-0 before:z-[-1]"
  >
    <div
      class="max-w-[1240px] mx-6 xl:mx-auto my-4 lg:w-2/3 flex flex-col gap-4"
    >
      <div
        class="text-white text-center"
      >
        <h2
          class="text-2xl md:text-3xl font-semibold text-white text-center uppercase mb-1"
        >
          Marvel Characters
        </h2>
        <p
          class="text-sm md:text-base text-center"
        >
          Get hooked on a hearty helping of heroes and villains from the humble House of Ideas!
        </p>
      </div>
      <form
        aria-label="Desktop Characters List Control Panel"
        class="focus-within gap-4 hidden md:flex justify-center flex-wrap"
      >
        <fieldset
          class="text-white grow"
        >
          <legend
            class="mb-1"
          >
            Search by name
          </legend>
          <input
            aria-label="search by name input"
            class="py-2 px-3 bg-black border border-white shadow appearance-none focus-visible-border w-full"
            placeholder="type a character name"
            type="text"
            value=""
          />
        </fieldset>
        <fieldset
          class="text-white grow"
        >
          <legend
            class="mb-1"
          >
            Order results
          </legend>
          <div
            class="h-full bg-black border shadow appearance-none focus-visible-border border-white w-full with-tailwind-b62m3t-container"
          >
            <span
              class="with-tailwind-1f43avz-a11yText-A11yText"
              id="react-select-4-live-region"
            />
            <span
              aria-atomic="false"
              aria-live="polite"
              aria-relevant="additions text"
              class="with-tailwind-1f43avz-a11yText-A11yText"
              role="log"
            />
            <div
              class="bg-black shadow appearance-none border-none with-tailwind-13cymwt-control"
            >
              <div
                class=" with-tailwind-1fdsijx-ValueContainer"
              >
                <div
                  class="text-white with-tailwind-1jqq78o-placeholder"
                  id="react-select-4-placeholder"
                >
                  by name (A/Z)
                </div>
                <div
                  class=" with-tailwind-qbdosj-Input"
                  data-value=""
                >
                  <input
                    aria-activedescendant=""
                    aria-autocomplete="list"
                    aria-describedby="react-select-4-placeholder"
                    aria-expanded="false"
                    aria-haspopup="true"
                    aria-label="order results select"
                    autocapitalize="none"
                    autocomplete="off"
                    autocorrect="off"
                    class=""
                    id="react-select-4-input"
                    role="combobox"
                    spellcheck="false"
                    style="opacity: 1; width: 100%; grid-area: 1 / 2; min-width: 2px; border: 0px; margin: 0px; outline: 0; padding: 0px;"
                    tabindex="0"
                    type="text"
                    value=""
                  />
                </div>
              </div>
              <div
                class=" with-tailwind-1hb7zxy-IndicatorsContainer"
              >
                <span
                  class=" with-tailwind-1u9des2-indicatorSeparator"
                />
                <div
                  aria-hidden="true"
                  class=" with-tailwind-1xc3v61-indicatorContainer"
                >
                  <svg
                    aria-hidden="true"
                    class="with-tailwind-tj5bde-Svg"
                    focusable="false"
                    height="20"
                    viewBox="0 0 20 20"
                    width="20"
                  >
                    <path
                      d="M4.516 7.548c0.436-0.446 1.043-0.481 1.576 0l3.908 3.747 3.908-3.747c0.533-0.481 1.141-0.446 1.574 0 0.436 0.445 0.408 1.197 0 1.615-0.406 0.418-4.695 4.502-4.695 4.502-0.217 0.223-0.502 0.335-0.787 0.335s-0.57-0.112-0.789-0.335c0 0-4.287-4.084-4.695-4.502s-0.436-1.17 0-1.615z"
                    />
                  </svg>
                </div>
              </div>
            </div>
          </div>
        </fieldset>
        <fieldset
          class="text-white w-full lg:w-auto"
        >
          <legend
            class="mb-1"
          >
            Filter results:
          </legend>
          <ul
            class="md:h-full flex flex-col md:flex-row gap-2 md:items-center"
          >
            <li
              class="flex items-center"
            >
              <input
                aria-label="withImage"
                class="mr-2 h-4 w-4 checked:bg-red bg-black border border-white shadow appearance-none focus-visible-border "
                id="withImage"
                name="withImage"
                type="checkbox"
                value="withImage"
              />
              <label
                for="withImage"
              >
                withImage
              </label>
            </li>
            <li
              class="flex items-center"
            >
              <input
                aria-label="withDescription"
                class="mr-2 h-4 w-4 checked:bg-red bg-black border border-white shadow appearance-none focus-visible-border "
                id="withDescription"
                name="withDescription"
                type="checkbox"
                value="withDescription"
              />
              <label
                for="withDescription"
              >
                withDescription
              </label>
            </li>
          </ul>
        </fieldset>
      </form>
    </div>
  </section>
</DocumentFragment>
`;

exports[`matches mobile snapshot 1`] = `
<DocumentFragment>
  <section
    class="relative flex flex-col items-center justify-around pt-logoDefaultHeight gap-2 min-h-64 md:min-h-96 lg:min-h-72 mb-4 bg-black bg-hero-image bg-center bg-cover before:absolute before:content[\\"\\"] before:bg-gradient-to-b before:from-trans-0.5-black before:from-50% before:left-0 before:top-0 before:z-[-1]"
  >
    <div
      class="max-w-[1240px] mx-6 xl:mx-auto my-4 lg:w-2/3 flex flex-col gap-4"
    >
      <div
        class="text-white text-center"
      >
        <h2
          class="text-2xl md:text-3xl font-semibold text-white text-center uppercase mb-1"
        >
          Marvel Characters
        </h2>
        <p
          class="text-sm md:text-base text-center"
        >
          Get hooked on a hearty helping of heroes and villains from the humble House of Ideas!
        </p>
      </div>
    </div>
  </section>
</DocumentFragment>
`;
